javaagent/sdk-migration-config.yaml

```diff
@@ -0,0 +1,291 @@
+            client_key: ${OTEL_EXPORTER_OTLP_TRACES_CLIENT_KEY}
+            # Configure mTLS client certificate.
+            client_certificate: ${OTEL_EXPORTER_OTLP_TRACES_CLIENT_CERTIFICATE}
+            # Configure compression.
+            compression: ${OTEL_EXPORTER_OTLP_TRACES_COMPRESSION:-gzip}
+            # Configure max time (in milliseconds) to wait for each export.
+            timeout: ${OTEL_EXPORTER_OTLP_TRACES_TIMEOUT:-10000}
+            # Configure headers. Entries have higher priority than entries from .headers_list.
+            headers: []
+            # Configure headers. Entries have lower priority than entries from .headers.
+            # The value is a list of comma separated key-value pairs matching the format of OTEL_EXPORTER_OTLP_HEADERS. See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#configuration-options for details.
+            headers_list: ${OTEL_EXPORTER_OTLP_TRACES_HEADERS}
+  # Configure span limits. See also attribute_limits.
+  limits:
+    # Configure max attribute value size. Overrides .attribute_limits.attribute_value_length_limit.
+    attribute_value_length_limit: ${OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT}
+    # Configure max attribute count. Overrides .attribute_limits.attribute_count_limit.
+    attribute_count_limit: ${OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT:-128}
+    # Configure max span event count.
+    event_count_limit: ${OTEL_SPAN_EVENT_COUNT_LIMIT:-128}
+    # Configure max span link count.
+    link_count_limit: ${OTEL_SPAN_LINK_COUNT_LIMIT:-128}
+    # Configure max attributes per span event.
+    event_attribute_count_limit: ${OTEL_EVENT_ATTRIBUTE_COUNT_LIMIT:-128}
+    # Configure max attributes per span link.
+    link_attribute_count_limit: ${OTEL_LINK_ATTRIBUTE_COUNT_LIMIT:-128}
+  # Configure the sampler.
+  sampler:
+    # Configure sampler to be parent_based.
+    parent_based:
+      # Configure root sampler.
+      root:
+        # Configure sampler to be always_on.
+        always_on: {}
+      # Configure remote_parent_sampled sampler.
+      remote_parent_sampled:
+        # Configure sampler to be always_on.
+        always_on: {}
+      # Configure remote_parent_not_sampled sampler.
+      remote_parent_not_sampled:
+        # Configure sampler to be always_off.
+        always_off: {}
+      # Configure local_parent_sampled sampler.
+      local_parent_sampled:
+        # Configure sampler to be always_on.
+        always_on: {}
+      # Configure local_parent_not_sampled sampler.
+      local_parent_not_sampled:
+        # Configure sampler to be always_off.
+        always_off: {}
+      # Uncomment to configure the rule_based_routing sampler from https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/samplers
+      # to drop spans to spring actuator endpoints.
+      # Configure the parent_based sampler's root sampler to be rule_based_routing sampler.
```

Comment:
why leave commented out?

Reply:
The thought was to leave the default parent based sampler config from `opentelemetry-config`, but also show the user how they might change the config to solve this use case.

Backing up a bit, I think we (the java SIG) need to maintain start template(s), based on the `opentelemetry-configuration` examples, but with the additional properties that java users will often care about. If we agree on that, where should those templates live and what should their contents be? Should this config file serve that purpose, either temporarily or permanently?

Reply:
I went ahead and set the root sampler to the rule based routing sampler by default. Called out this difference from the standard `sdk-config.yaml` template in comments at the top of the file. I figure its such an important use case that its useful to highlight.